api/src/wfl/module/all.clj

```diff
@@ -138,5 +141,19 @@
 (s/def ::name string?)
 (s/def ::methodConfiguration (s/and string? util/terra-namespaced-name?))
 (s/def ::methodConfigurationVersion integer?)
-(s/def ::watchers (s/* util/email-address?))
+(s/def ::email util/email-address?)
+(s/def ::legacyWatcher ::email)
+(s/def ::taggedWatcher (s/or :email slack/email-watcher?
+                             :slack slack/slack-channel-watcher?))
+(s/def ::watcher
+  (s/or :legacy_watcher  ::legacyWatcher
+        :tagged_watcher  ::taggedWatcher))
+(s/def ::watchers (s/* ::watcher))
 (s/def ::workspace (s/and string? util/terra-namespaced-name?))
+
+(defn parse-watcher
+  "Parse watchers based on specs."
+  [watcher]
+  (if (s/valid? ::legacyWatcher watcher)
+    watcher
+    (edn/read-string watcher)))
```

Comment:
Use `conform`?
```clojure
  (s/conform ::watcher ["EmailAddress" "the-mgt@omf.org"])
  ;; => [:tagged_watcher [:email ["EmailAddress" "the-mgt@omf.org"]]]
  (s/conform ::watcher ["SlackChannel" "Fnord"])
  ;; => :clojure.spec.alpha/invalid
  (s/conform ::watcher ["SlackChannel" "ChannelOfBob"])
  ;; => [:tagged_watcher [:slack ["SlackChannel" "ChannelOfBob"]]]
  (s/conform ::watcher "the-mgt@omf.org")
  ;; => [:legacy_watcher "the-mgt@omf.org"]
```
That will even tell you how to interpret each watcher spec.

Reply:
TIL

Reply:
TIL, but I'm not sure what extra value `conform` brings here than `s/valid?`, the deserializer will need to understand both legacy and new watchers, and in reality it deals with stuff like:

```
["hornet@broadinstitute.org"
"[\"EmailAddress\" \"hornet@broadinstitute.org\"]"
"[\"SlackChannel\" \"C026PTM4XPA\"]"]
```
the `if` will binary classify them into legacy: `"hornet@broadinstitute.org"` and tagged (but edn-fied) watcher: `"[\"EmailAddress\" \"hornet@broadinstitute.org\"]"` and `"[\"SlackChannel\" \"C026PTM4XPA\"]"`, how does conform validate the latter without trying to parse them first?